src/makeInfinityAdder.js

```diff
@@ -1,10 +1,35 @@
 'use strict';
 
 /***
+ *
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let clearCall = 0;
+  let result = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      clearCall++;
+
+      if (clearCall === 1) {
+        return result;
+      }
+
+      if (clearCall === 2) {
+        result = 0;
+
+        return result;
+      }
+    }
+
+    result += arg;
```

Comment:
The code does not handle the case when a non-number argument is passed to the 'adder' function. It's important to ensure that only numbers are added to the 'result' to avoid unexpected behavior.